flexget/components/ftp/sftp_client.py

```diff
@@ -0,0 +1,423 @@
+
+    def get_dir_handler(self, get_size: bool, files_only: bool, entry_accumulator: list) -> Callable[[str], None]:
+        """
+        Builds a file node handler suitable for use with pysftp.Connection.walktree
+
+        :param get_size: boolean indicating whether to compute the for each file
+        :param files_only: Boolean indicating whether to skip directories
+        :param entry_accumulator: list to add entries to
+        """
+        return partial(Handlers.handle_directory, self._sftp, self._logger, self._prefix, get_size, files_only,
+                       entry_accumulator)
+
+    def get_unknown_handler(self) -> Callable[[str], None]:
+        """
+        Builds an unknown node handler suitable for use with pysftp.Connection.walktree
+        """
+        return partial(Handlers.handle_unknown, self._logger)
+
+    def get_null_handler(self) -> Callable[[str], None]:
+        """
+        Builds a noop node handler suitable for use with pysftp.Connection.walktree
+        """
+        return partial(Handlers.null_node_handler, self._logger)
+
+
+class Handlers:
+    @classmethod
+    def handle_file(cls, sftp: 'pysftp.Connection', logger: logging.Logger, prefix,get_size,
+                    entry_accumulator, path):
+        """
+        File node handler. Adds a file entry to entry_accumulator.
+
+        :param sftp: A pysftp.Connection object
+        :param logger: a logger object
+        :param prefix: SFTP URL prefix
+        :param get_size: boolean indicating whether to compute the size of each file
+        :param entry_accumulator: a list in which to store entries
+        :param path: path to handle
+        """
+        size_handler = partial(cls._file_size, sftp)
+        entry = cls._get_entry(sftp, logger, prefix, size_handler, get_size, path)
+        entry_accumulator.append(entry)
+
+    @classmethod
+    def handle_directory(cls, sftp: 'pysftp.Connection', logger: logging.Logger, prefix: str, get_size: bool,
+                         files_only: bool, entry_accumulator: List, path: str):
+        """
+        Directory node handler. Adds a directory entry to entry_accumulator.
+
+        :param sftp: A pysftp.Connection object
+        :param logger: a logger object
+        :param prefix: SFTP URL prefix
+        :param get_size: boolean indicating whether to compute the size of each directory
+        :param files_only: Boolean indicating whether to skip directories
+        :param entry_accumulator: a list in which to store entries
+        :param path: path to handle
+        """
+        if files_only:
+            return
+
+        dir_size: Callable[[str], int] = partial(cls._dir_size, sftp)
+        entry: Entry = cls._get_entry(sftp, logger, prefix, dir_size, get_size, path)
+        entry_accumulator.append(entry)
+
+    @staticmethod
+    def handle_unknown(logger: logging.Logger, path: str):
```

Comment:
i don't see a need to pass the logger around to these methods. Using the module level logger is perfectly fine IMO